crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs

```diff
@@ -275,32 +300,65 @@ struct UpdateAgentInfoInput<'borrow> {
     expires_after: u64,
     space: Arc<KitsuneSpace>,
     agent: Arc<KitsuneAgent>,
+    arc: DhtArc,
     urls: &'borrow Vec<TxUrl>,
     evt_sender: &'borrow futures::channel::mpsc::Sender<KitsuneP2pEvent>,
+    internal_sender: &'borrow ghost_actor::GhostSender<SpaceInternal>,
     network_type: NetworkType,
     mdns_handles: &'borrow mut HashMap<Vec<u8>, Arc<AtomicBool>>,
     bootstrap_service: &'borrow Option<Url2>,
 }
 
+#[cfg(feature = "sharded")]
+async fn update_arc_length(
+    evt_sender: &futures::channel::mpsc::Sender<KitsuneP2pEvent>,
+    space: Arc<KitsuneSpace>,
+    arc: &mut DhtArc,
+) -> KitsuneP2pResult<()> {
+    // TODO: Get the peer density and update the arc if feature flag is set.
+    let density = evt_sender
+        .query_peer_density(space.clone(), arc.clone())
+        .await?;
+    arc.update_length(density);
+    Ok(())
```

Comment:
Seems like this TODO is done?

```suggestion
    let density = evt_sender
        .query_peer_density(space.clone(), arc.clone())
        .await?;
    arc.update_length(density);
    Ok(())
```